lib/rules/no-control-regex.js

```diff
@@ -51,14 +51,43 @@ module.exports = {
             return null;
         }
 
+        /**
+         * Check the regex expression
+         * @param {String} regexStr regex as string to check
+         * @returns {Boolean} returns true if finds control characters on regex
+         * @private
+         */
+        function checkRegExp(regexStr) {
```

Comment:
This is a bad name and the jsdoc has a bad description. How about `hasControlChars`? Protip: when choosing a name for a function, look to the returned expression for suggestions.